File: octave/vector_ad_xam.m

% This file can be automatically generaeted using the following command
% m4 ../octave.m4 ../xam/vector_ad_xam.m4 > vector_ad_xam.m
% -----------------------------------------------------------------------------
%         cppad_swig: A C++ Object Library and SWIG Interface to CppAD
%          Copyright (C) 2017-17 Bradley M. Bell (bradbell@seanet.com)
%              This program is distributed under the terms of the
%          GNU Affero General Public License version 3.0 or later see
%                     http://www.gnu.org/licenses/agpl.txt
% -----------------------------------------------------------------------------
% std::vector<a_double>
% -----------------------------------------------------------------------------
function ok = vector_ad_xam()
	%
	% load the cppad swig library
	m_cppad
	%
	% initialize return variable
	ok = true;
	n = 4;
	a_vec = m_cppad.vector_ad(n);
	%
	% check size
	ok = ok && a_vec.size() == n;
	%
	% setting elements
	for i = [ 0 :(n -1) ]
		ad = m_cppad.a_double(2.0 * i);
		a_vec(i) = ad;
	end
	% getting elements
	for i = [ 0 :(n -1) ]
		a_element = a_vec(i);
		ok = ok && a_element.value() == 2.0 * i;
	end
	return;
end

File: lib/example/cplusplus/vector_ad_xam.cpp
// This file can be automatically generaeted using the following command
// m4 ../../xam/cplusplus.m4 ../../xam/vector_ad_xam.xam > vector_ad_xam.cpp
// -----------------------------------------------------------------------------
//         cppad_swig: A C++ Object Library and Swig Interface to Cppad
//          Copyright (C) 2017-17 Bradley M. Bell (bradbell@seanet.com)
//              This program is distributed under the terms of the
//          GNU Affero General Public License version 3.0 or later see
//                     http://www.gnu.org/licenses/agpl.txt
// -----------------------------------------------------------------------------
// std::vector<a_double>
// -----------------------------------------------------------------------------
# include <cstdio>
# include <cppad/swig/a_double.hpp>
# include <cppad/swig/a_fun.hpp>
# include <cppad/swig/function.hpp>
# include <cppad/swig/typedef.hpp>

bool vector_ad_xam(void) {
	using cppad_swig::a_double;
	//
	// initialize return variable
	bool ok = true;
	size_t n = 4;
	cppad_swig::vector_ad a_vec = cppad_swig::vector_ad(n);
	//
	// check size
	ok = ok && a_vec.size() == n;
	//
	// setting elements
	for(size_t i = 0; i < n ; i++) {
		cppad_swig::a_double ad = cppad_swig::a_double(2.0 * i);
		a_vec[i] = ad;
	}
	// getting elements
	for(size_t i = 0; i < n ; i++) {
		cppad_swig::a_double a_element = a_vec[i];
		ok = ok && a_element.value() == 2.0 * i;
	}
	return( ok );
}
